### PR TITLE
Continuation of kernel_info_request message handling

### PR DESCRIPTION
**Pull request by mariusvniekerk**
This continues the work done in #51

### CODE REVIEW

File: kernel-api/src/main/scala/org/apache/toree/interpreter/Interpreter.scala [hunk 1]
```diff
@@ -23,6 +23,12 @@ import org.apache.toree.kernel.api.KernelLike
 
 import scala.tools.nsc.interpreter._
 
+case class LanguageInfo(
+                 name: String,
+                 version: String,
+                 fileExtension: Option[String] = None,
+                 pygmentsLexer: Option[String] = None) {
+}
```
Comment (rgbkrk) [inline kernel-api/src/main/scala/org/apache/toree/interpreter/Interpreter.scala]: 👍

Comment (rgbkrk) [inline kernel-api/src/main/scala/org/apache/toree/interpreter/Interpreter.scala]: Yeah we certainly use the language info, including it in the metadata in the notebook document. Same as jupyter/notebook.

**Comment by mariusvniekerk** [comment]
@lbustelo this is ready for review now.

File: pyspark-interpreter/src/main/scala/org/apache/toree/kernel/interpreter/pyspark/PySparkInterpreter.scala [hunk 2]
```diff
@@ -148,4 +154,19 @@ class PySparkInterpreter(
   // Unsupported
   override def doQuietly[T](body: => T): T = ???
 
+  override def languageInfo: LanguageInfo = {
+    if ((!pySparkService.isRunning) || (!pySparkState.isReady)) {
+      LanguageInfo(
+        "python",
+        version = "UNKNOWN",
```
Comment (lbustelo) [inline pyspark-interpreter/src/main/scala/org/apache/toree/kernel/interpreter/pyspark/PySparkInterpreter.scala]: Really no way of knowing python version here? Is Jupyter fine with "Unknown"?

Comment (mariusvniekerk) [inline pyspark-interpreter/src/main/scala/org/apache/toree/kernel/interpreter/pyspark/PySparkInterpreter.scala]: I guess we can subprocess call python before we initialize it to get a --version?

**Comment by mariusvniekerk** [comment]
@lbustelo round 2

**Pull request merged**